### PR TITLE
1837 add background job arm template

### PR DESCRIPTION
**Pull request by rizzkhan1**
### Context
Automate deployment of background jobs docker images onto an Azure Container Instance (ACI). ARM template has been created to deploy the ACI infrastructure using Azure Devops Release pipeline. 

### Changes proposed in this pull request
New additional ARM template for for background job creation. This ARM template has no dependency and does not replace existing template for Backend app services (template.json). 

### Guidance to review
New file created under Azure folder

### CODE REVIEW

**Comment by defong** [comment]
do you need all those parameters

**Comment by EwanNoble** [comment]
Secrets should probably be passed in as `securestring`s

**Comment by rizzkhan1** [comment]
> do you need all those parameters
No, only 3 parameters are required to create an Azure container instance. Most of the parameters are for the application environment variables. 

> Secrets should probably be passed in as `securestring`s
Secret are being passed through pipeline variables in a clear text and are converted to secure values under environmental variables section (not parameter). This is identical to all other web apps application settings. Using secure string will double encrypt the secret and application may not support that.

**Comment by misaka** [comment]
> > do you need all those parameters
> 
> No, only 3 parameters are required to create an Azure container instance. Most of the parameters are for the application environment variables.

We've gotten all of these because we've done a wholesale copy of the `mcbe` app ... which kind of makes sense, although:

1. most of these aren't (probably) won't be needed, so we could reduce clutter by stripping them out
2. for the ones that are needed, if they ever change we'll need to ensure we update both apps

We could get two birds stoned at once if we found a way to re-use values between `mcbe` and this app, how much work would that be?

**Comment by rizzkhan1** [comment]
> 1. most of these aren't (probably) won't be needed, so we could reduce clutter by stripping them out 
> 2. for the ones that are needed, if they ever change we'll need to ensure we update both apps

Happy to declutter them once we know which ones aren't needed.
I could merge mcbe and mcbg arm templates so we won't have to change values in multiple places but then we lose lots of flexibility in terms of release pipeline and deployments.

**Comment by misaka** [comment]
@rizzkhan1 Do we take the values for these params from a central place that is shared with `mcbe` at least? That would probably be good enough, in which case I'm not sure I mind the clutter for the sake of expediency.

**Comment by rizzkhan1** [comment]
@misaka values are passed from AzureDevops pipelines, most of them have been grouped so they are quite easy to manage.

File: azure/backgroundJobTemplate.json [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+            "value": "[parameters('imageName')]"
+          },
+          "environmentVariables": {
+            "value": [
+              {
+                "name": "RAILS_ENV",
+                "value": "[parameters('railsEnv')]"
+              },
+              {
+                "name": "AUTHENTICATION_TOKEN",
```
Comment (misaka) [inline azure/backgroundJobTemplate.json]: I just had a look through the code for how this is used, and it's only used in authenticating requests for API V1. If we remove this then I *think* the authentication token will be blank, and while we don't expose a way to reach API V1 here, if we ever did using this configuration we'd be leaking data.

At the risk of being paranoid, I say we just leave it here. What we really outta be doing is setting this kind of thing to a secure string (read: random) by default, with overrides per environment.

File: azure/backgroundJobTemplate.json [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+              {
+                "name": "SECRET_KEY_BASE",
+                "secureValue": "[parameters('secretKeyBase')]"
+              },
+              {
+                "name": "SENTRY_DSN",
+                "secureValue": "[parameters('sentryDSN')]"
+              },
+              {
+                "name": "SETTINGS__AUTHENTICATION__SECRET",
```
Comment (misaka) [inline azure/backgroundJobTemplate.json]: This one, like `AUTHENTICATION_TOKEN`, is used for communicating with an API ... V2 in this case. Like that one, I say we leave this one until we have it set to a reasonably secure default.

File: azure/backgroundJobTemplate.json [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+                "secureValue": "[parameters('settingsAuthenticationSecret')]"
+              },
+              {
+                "name": "SETTINGS__MANAGE_API__SECRET",
```
Comment (misaka) [inline azure/backgroundJobTemplate.json]: This is used when communicating with `manage-courses-api`. I believe the only outgoing connection we'll make in this app will be to search-and-compare so this is one we can probably get rid of.

File: azure/backgroundJobTemplate.json (hunk 1 above)
Comment (rizzkhan1) [reply azure/backgroundJobTemplate.json]: We are already using this configuration in rest of the other ARM templates. Authentication token value goes in application settings which takes only string values.

**Pull request merged**